### ISSUE TITLE
CLI: Implement Helm flags on server install

### ISSUE THREAD

**Issue by vitorfhc** [hashicorp/waypoint#4403]
**Is your feature request related to a problem? Please describe.**

I was trying to install a server, and this was the output:

```
$ waypoint install -platform=kubernetes -accept-tos
❌ Installing Waypoint Helm chart...
! Error installing server into kubernetes: timed out waiting for the condition
```

The server was deployed successfully even though the installation timed out. It needs to be more accurate.

If I try to rerun the command, it fails because there's a Helm release with the same name. Then you start questioning if it failed or not.

**Describe the solution you'd like**

It would be nice to control Helm flags (e.g., `--atomic` or `--wait`). It would be even better to set `atomic` to `true` by default, so we avoid misleading errors. If it fails for any reason, then we uninstall it.

Maybe we could even detect if the error was a timeout and suggest the user increase it using `-helm-timeout`.

**Describe alternatives you've considered**

We could increase the timeout for Helm action ([code](https://github.com/hashicorp/waypoint/blob/c19bb81db05e5925902ba71bcb7b74055df02b8d/internal/serverinstall/k8s.go#L161)), but I don't think it's a good solution.

**Explain any additional use-cases**

There are lots of reasons to customize the Helm behaviour. Someone may want to add a description, use the `--replace`, or skip the CRDs.

**Additional context**

It would be tiring to create each Helm flag in the `install` command. We could do it dynamically.

Any `-helm-<flag>` flag would become a `client.<flag> = flag_value`.

Or we could do it at least for some flags like `--atomic` and `--timeout` at first.

**Comment by demophoon** [comment]
@vitorfhc, Thanks for the report!
We think that the error given is very vague and definitely unhelpful, there may be a way for us to inspect help after an error is thrown when installing using the kubernetes platform that we could throw here. Perhaps adding additional pre-install checks to make sure we can actually install and if not throwing a better error here.

**Comment by vitorfhc** [comment]
@demophoon yep, but don't you agree to set the `atomic` flag? Guaranteeing safe installs and upgrades makes sense.
What about the `-k8s-helm-timeout` flag I created in #4404?

It hangs and times out because the service never gets an external IP since I was using K3s. There isn't a way to check if this will happen. Having pre-install checks is nice, but it looks like a lot of effort, IMHO.
